src/peergos/server/tests/QuotaTests.java

```diff
@@ -0,0 +1,85 @@
+package peergos.server.tests;
+
+import org.junit.*;
+import org.junit.runner.*;
+import org.junit.runners.*;
+import peergos.server.*;
+import peergos.shared.*;
+import peergos.shared.crypto.*;
+import peergos.shared.user.*;
+import peergos.shared.user.fs.*;
+import peergos.shared.util.*;
+
+import java.lang.reflect.*;
+import java.net.*;
+import java.nio.file.*;
+import java.util.*;
+
+import static peergos.server.tests.UserTests.ensureSignedUp;
+
+@RunWith(Parameterized.class)
+public class QuotaTests {
+
+    public static int RANDOM_SEED = 666;
+    private final NetworkAccess network;
+    private final Crypto crypto = Crypto.initJava();
+
+    private static Random random = new Random(RANDOM_SEED);
+
+    public QuotaTests(String useIPFS, Random r) throws Exception {
+        int portMin = 9000;
```

Comment:
If you pass 0 as the port ``ServerSocket`` will give you an available port. That would be safer.

Reply:
How do I then get the port to use elsewhere?

Reply:
expose it in the service.

Reply:
Expose it how? I don't have a pointer to the service here and the only way to get one would to give your Command class a return value, but I don't see how that would work in the general Command structure.